packages/react/src/components/checkbox/checkbox.react.stories.tsx

```diff
@@ -0,0 +1,81 @@
+import { Checkbox, CheckboxProps, HelperText } from '@onfido/castor-react';
+import React, { ChangeEvent, useRef } from 'react';
+import { Meta, omit, Story, storyOf } from '../../../../../docs';
+
+export default {
+  title: 'React/Checkbox',
+  component: Checkbox,
+  argTypes: {
+    ...omit<CheckboxProps>('className', 'style'),
+    children: { control: 'text' },
+    bordered: {
+      table: { type: { summary: 'boolean' } },
+    },
+    invalid: {
+      table: { type: { summary: 'boolean' } },
+    },
+    disabled: {
+      table: { type: { summary: 'boolean' } },
+    },
+  },
+  args: {
+    bordered: false,
+    invalid: false,
+    disabled: false,
+  },
+} as Meta<CheckboxProps>;
+
+export const Playground: Story<CheckboxProps> = (props: CheckboxProps) => (
+  <Checkbox {...props} />
+);
+
+export const Bordered = storyOf(Checkbox, 'bordered', [true, false], {
+  labelMode: 'children',
+});
+Bordered.argTypes = omit<CheckboxProps>('bordered');
+Bordered.args = { name: 'bordered-playground' };
+
+export const Invalid = storyOf(Checkbox, 'invalid', [true, false], {
+  labelMode: 'children',
+});
+Invalid.argTypes = omit<CheckboxProps>('invalid');
+Invalid.args = { name: 'invalid-playground' };
+
+export const Disabled = storyOf(Checkbox, 'disabled', [true, false], {
+  labelMode: 'children',
+});
+Disabled.argTypes = omit<CheckboxProps>('disabled');
+Disabled.args = { name: 'disabled-playground' };
+
+export const AsIndeterminate = (props: CheckboxProps) => {
+  const checkboxRef = useRef<HTMLInputElement | null>(null);
+
+  const handleChange = ({
+    target: { checked: indeterminate },
+  }: ChangeEvent<HTMLInputElement>) => {
+    if (checkboxRef.current) checkboxRef.current.indeterminate = indeterminate;
+  };
+
+  return <Checkbox {...props} ref={checkboxRef} onChange={handleChange} />;
+};
```

Comment:
I found it unintuitive that I had to click the checkbox to make the indeterminate state display.
I was expecting it to render in that state from the beginning.

Also, we are supposed to use these for visual regression testing later.

I'd recommend using `storyOf('indeterminate', [true, false])` instead.

Reply:
I'm not sure about this.

We don't render stories for "checked" also.

Not sure "true" for such and then "false" would make sense, because you can still change the state, so when changed the label would not properly indicate the correct state. It could then disallow the state change, but also would be odd if interacting with an element does not do as it should.

That's why the story is called "as" - it does not demonstrate differences between one and another, but instead shows the state of such when it's enabled, that is when you "check". This same applies to Button component, that has a story "As Anchor".

Reply:
Maybe we should have `checked` stories?
It'll be clearer when we do have those visual tests.

The way I think about it is "`storiesOf` shows the default state when first rendered, but still allows interactivity".

Something we could do is controlled inputs, with a forced `checked` which means you can't change its state.
I wouldn't mind that if you prefer it.

Reply:
I actually disagree here...

O think `storyOf` should demonstrate prop variation.

`checked` and `indeterminate` is not part of props. Well, "checked" sort of is, if you decide to go the controlled way. But "indeterminate" is not - the code demonstrates only way this can be enabled via code.

But I do agree having state stories is valuable for visual tests, but maybe that's another bigger taks? For example, such would also be needed for Button's "pressed" state.

Reply:
I see what you're saying, they're props to React but not specifically our component.
Still, we have different styles for each of those states.

Yea perhaps visual tests can be done for HTML + CSS, but it's also valuable to check that `<Checkbox indeterminate />` renders with the correct CSS.

Alright nevermind, this seems like a different and bigger discussion, let's leave as is for now.

Reply:
So yet `<Checkbox indeterminate />` will not work - you can't control this state like such...

Reply:
You're right, that to me is a bug though, React should control DOM properties when there's no attribute equivalent.

Apparently they are not supporting it because it can't be server-rendered https://github.com/facebook/react/issues/1798#issuecomment-333414857.

We should probably specify that prop and implement a `useEffect` hook to switch its `indeterminate` state.
Back to merging `ref`s fun.

As a Castor consumer I'd totally expect to be able to do `<Checkbox indeterminate />`